includes/class.llms.ajax.handler.php

```diff
@@ -774,12 +774,11 @@ public static function quiz_answer_question( $request ) {
 	 * @return array
 	 */
 	public static function quiz_end( $request, $attempt = null ) {
-
-		$err = new WP_Error();
+		$err     = new WP_Error();
+		$student = llms_get_student();
```

Comment:
the code above has a check on whether the student exists, this probably means that at this stage there can be the possibility of having a `null` student here. But you're going to use the student instance on line 808 without checks... I'd say move also:
```
			if ( ! $student ) {
				$err->add( 400, __( 'You must be logged in to take quizzes.', 'lifterlms' ) );
				return $err;
			}
```
outside of the `if ( ! $attempt ) {`?

Reply:
oh, right. good catch 👍